assets/admin/emails/email-preview-button/email-preview-button.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { useState, createPortal } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { store as editorStore } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * The email preview button component.
+ *
+ * @param {Object}  props
+ * @param {string}  props.previewLink    The preview link.
+ * @param {number}  props.postId         The post ID.
+ * @param {boolean} props.isSaveable     If the post is savable.
+ * @param {boolean} props.isAutosaveable If the post can be autosaved.
+ * @param {boolean} props.isLocked       If the post is locked.
+ * @param {boolean} props.isDraft        If the post is a draft.
+ * @param {Object}  props.autosave       The autosave action.
+ * @param {Object}  props.savePost       The save post action.
+ *
+ * @return {Object|null} A portal element or null if the container is not available.
+ */
+export const EmailPreviewButton = ( {
+	previewLink,
+	postId,
+	isSaveable,
+	isAutosaveable,
+	isLocked,
+	isDraft,
+	autosave,
+	savePost,
+} ) => {
+	const [ isSaving, setIsSaving ] = useState( false );
+	const container = document.querySelector(
+		'.block-editor-post-preview__dropdown'
+	);
+	if ( ! container ) {
+		return null;
+	}
+
+	/**
+	 * Open the preview in a new window. Triggers an autosave when needed.
+	 *
+	 * @param {MouseEvent} event
+	 * @return {Promise<void>}
+	 */
+	const openPreviewWindow = async ( event ) => {
+		event.preventDefault();
+
+		if ( isAutosaveable && ! isLocked ) {
+			setIsSaving( true );
+
+			if ( isDraft ) {
+				await savePost( { isPreview: true } );
+			} else {
+				await autosave( { isPreview: true } );
+			}
+
+			setIsSaving( false );
+		}
+
+		window.open( event.target.href, 'sensei-email-preview-' + postId );
+	};
+
+	return createPortal(
+		<Button
+			href={ previewLink }
+			onClick={ openPreviewWindow }
+			isBusy={ isSaving }
+			disabled={ ! isSaveable || isSaving }
+			className="sensei-email-preview-button"
+			variant="tertiary"
+			role="menuitem"
+		>
+			{ __( 'Preview', 'sensei-lms' ) }
+		</Button>,
+		container
+	);
+};
+
+export default compose( [
```

Comment:
Can we replace with the hook version, like useSelect and useDispachet.

The HOC is outdated because it messes a lot with the rendering tree with a lot of necessary nesting.

Reply:
I did the update in https://github.com/Automattic/sensei/pull/6583/commits/3c337275efd632c43f0476b376374f1477426cab but please let me know if something is a miss.

Really appreciate the React help!